### ISSUE TITLE
Copying object with new storage class does not appear to save storage class

### ISSUE THREAD

**Issue by a-farrukh** [adobe/S3Mock#1429]
Hi,

I am using an S3 mock test containers setup and it has been wonderful.

However, I ran into an issue copying objects with a different storage class.
It successfully copies the object, but the new storage class does not have an effect.

I ran the same code plugging in real S3 endpoints and bucket/keys and it worked, but using the S3 mock with a modified test containers endpoint, it did not seem to have an effect, though the object was copied.

```
    private CopyObjectRequest createRequest(String bucket, String key) {
        return CopyObjectRequest.builder()
                .storageClass(storageClass)
                .sourceBucket(bucket)
                .sourceKey(key)
                .destinationBucket(bucket)
                .destinationKey(key)
                .build();
    }
```

This is the request I am creating, and we simply use `S3AsyncClient` and override the endpoint to use that of the test container.
Copying to a different location e.g. by changing the `destinationKey` to `"updated" + key` has the same issue.

`storageClass` is `StorageClass.STANDARD_IA`

Version is `2.17.0`

**Comment by afranken** [comment]
S3Mock is currently not persisting the StorageClass at all - we just return `STANDARD` for all requests.
I'll look into supporting StorageClass persistence.

**Comment by a-farrukh** [comment]
@afranken 
Hey Arne,

Any ideas for a timeframe for when this can be implemented?

Thanks!

**Comment by afranken** [comment]
@a-farrukh thanks for the reminder. I've been busy with other work but I started working on this now.
I'll have to ask you for a bit more patience, I hope to release a new version including a fix within the next two weeks or so.

I still have to look into the TODOs, and also other APIs to support - as I wrote in my last comment we do not support StorageClass persistence at all right now, we just return `STANDARD` in every response.

**Comment by a-farrukh** [comment]
> @a-farrukh thanks for the reminder. I've been busy with other work but I started working on this now. I'll have to ask you for a bit more patience, I hope to release a new version including a fix within the next two weeks or so.
> 
> I still have to look into the TODOs, and also other APIs to support - as I wrote in my last comment we do not support StorageClass persistence at all right now, we just return `STANDARD` in every response.

Thanks for your response.

No problem! Do take as much time as you need to make sure it all works well and good :-)

**Comment by afranken** [comment]
@a-farrukh I released 3.4.0 with support for StorageClass attributes. :)
